src/path.rs

```diff
@@ -595,4 +666,64 @@ mod tests {
             "Normal(abc)"
         );
     }
+
+    #[test]
+    fn test_path_components() {
+        let p = Path::new("");
+        let c: Vec<_> = p.components().collect();
+        assert!(c.is_empty());
+
+        let p = Path::new("/");
+        let c: Vec<_> = p.components().collect();
+        assert_eq!(c, [Component::RootDir]);
+
+        let p = Path::new("/ab/cd/ef/..");
+        let c: Vec<_> = p.components().collect();
+        assert_eq!(
+            c,
+            [
+                Component::RootDir,
+                Component::normal("ab").unwrap(),
+                Component::normal("cd").unwrap(),
+                Component::normal("ef").unwrap(),
+                Component::ParentDir,
+            ]
+        );
+
+        let p = Path::new("ab/cd/ef");
+        let c: Vec<_> = p.components().collect();
+        assert_eq!(
+            c,
+            [
+                Component::normal("ab").unwrap(),
+                Component::normal("cd").unwrap(),
+                Component::normal("ef").unwrap(),
+            ]
+        );
+
+        let p = Path::new("///ab///cd///ef//");
+        let c: Vec<_> = p.components().collect();
+        assert_eq!(
+            c,
+            [
+                Component::RootDir,
+                Component::normal("ab").unwrap(),
+                Component::normal("cd").unwrap(),
+                Component::normal("ef").unwrap(),
+            ]
+        );
+
+        // PathBuf uses the same implementation, just do one test to
+        // verify.
+        let p = PathBuf::new("/a/b");
+        let c: Vec<_> = p.components().collect();
+        assert_eq!(
+            c,
+            [
+                Component::RootDir,
+                Component::normal("a").unwrap(),
+                Component::normal("b").unwrap(),
+            ]
+        );
```

Comment:
Similarly: tests for CurDir? Even if we want `.` to be returned as a `Normal` I feel like there should be tests to express that.

Reply:
Added